packages/diagrams/frontend/sirius-components-diagrams/src/renderer/dropNode/useDropNode.ts

```diff
@@ -72,6 +72,20 @@ const getNodeDepth = (node: Node<NodeData>, intersections: Node<NodeData>[]): nu
   return nodeDepth;
 };
 
+const evaluateAbsolutePosition = (node: Node, nodes: Node[]): XYPosition => {
+  let nextParentId: string | undefined = node.parentId;
+  const positionAbsolute: XYPosition = { ...node.position };
+  while (nextParentId) {
+    const parent = nodes.find((n) => n.id === nextParentId);
```

Comment:
You should pass `isDescendantOf` to find the node like it's done with the `isDescendantOf` method. Otherwise it will have an impact on performance

Reply:
Done